### PR TITLE
feat(deployment): make resources configurable

### PR DESCRIPTION
**Pull request by ChrisRousey**
this PR makes resource limits and requests for deployments configurable and lowers our default value.

refs: #20, #15

### CODE REVIEW

**Review by flippingbits — COMMENTED** [review]
Looks good to me, thanks a lot :)

Only thing I would change is making the configuration 100% compatible with [Kubernetes](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-requests-and-limits-of-pod-and-container), .i.e., instead of `datacater.deployment.resources.cpu.request` we would have `datacater.deployment.resources.requests.cpu`.

This way we help users that are new to DataCater but have experience with Kubernetes.

**Comment by ChrisRousey** [comment]
> Only thing I would change is making the configuration 100% compatible with [Kubernetes](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-requests-and-limits-of-pod-and-container), .i.e., instead of `datacater.deployment.resources.cpu.request` we would have `datacater.deployment.resources.requests.cpu`.

Good point, thanks! I just pushed the change :)

File: platform-api/src/main/java/io/datacater/core/deployment/StaticConfig.java [hunk 1]
```diff
@@ -112,6 +117,19 @@ private EnvironmentVariables() {}
         ConfigProvider.getConfig()
             .getOptionalValue("datacater.pythonrunner.image.containerPort", Integer.class)
             .orElse(50000);
+
+    static final String DEPLOYMENT_RESOURCES_REQUEST_MEMORY =
+        ConfigProvider.getConfig()
+            .getOptionalValue("datacater.deployment.resources.request.memory", String.class)
```
Comment (flippingbits) [inline platform-api/src/main/java/io/datacater/core/deployment/StaticConfig.java:123]: Can you please make `request` plural, .i.e., `requests`, to match the `application.yml`?

**Review by flippingbits — COMMENTED** [review]
Thanks :) Found one small issue

**Review by flippingbits — APPROVED** [review]
Thanks, LGTM 💯

**Pull request merged**